lib/Service/ItemServiceV2.php

```diff
@@ -391,4 +429,62 @@ public function findAllWithFilters(
     ): array {
         return $this->mapper->findAllItems($userId, $type, $limit, $offset, $oldestFirst, $search);
     }
+
+
+    /**
+     * Share an item with a user
+     *
+     * @param string $userId Item owner
+     * @param int    $id     Item ID
+     * @param bool   $shareWithId   User to share with
+     *
+     * Sharing by copying - the item is duplicated, and the 'sharedBy' and
+     * 'sharedWith' fields are filled accordingly.
+     * We copy the 'feedId', because the article will still be owned by
+     * $userId, and it'll be stored in his feed
+     *
+     * @return Item
+     * @throws ServiceNotFoundException|ServiceConflictException
+     */
+    public function share(string $userId, int $id, string $shareWithId): Entity
+    {
+        // find item to share
+        try {
+            $item = $this->find($userId, $id);
+        } catch (DoesNotExistException $ex) {
+            throw ServiceNotFoundException::from($ex);
+        }
+
+        // duplicate the item
+        $sharedItem = Item::fromImport($item->jsonSerialize());
+
+        // copy and initialize fields
+        $sharedItem->setUnread(true);
+        $sharedItem->setStarred(false);
+        $sharedItem->setFeedId($item->getFeedId());
+        $sharedItem->setFingerprint($item->getFingerprint());
+        $sharedItem->setContentHash($item->getContentHash());
+        $sharedItem->setSearchIndex($item->getSearchIndex());
+
+        // set share data
+        $sharedItem->setSharedBy($userId);
+        $sharedItem->setSharedWith($shareWithId);
```

Comment:
In most cases yes, but not always.
Say that user A shares an article with user B. In this case it would work because the feed belongs to A.

However, if user B decides to share that same article with user C, there would be a problem. The owner of the article is still user A, so C would think that A shared it with them.

feed owner = article owner
shared_by = sender
shared_with = recipient

Reply:
If allowing to re-share a shared article the long-term consideration gets even more important.

Following scenario: User1 shares with User2, who shares with User3 after 14 days ...
- U1 now deletes the feed after 14.5 days which removes all of the shared items but U3 only had like 12 hours to read it and knew that U2 shared it.
- U3 could now share with U1 since they didn't knew that it started from them (causing the item to be duplicated for U1)